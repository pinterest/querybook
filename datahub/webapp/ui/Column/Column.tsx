import styled from 'styled-components';

export const Columns = styled.div.attrs({
    className: 'Columns',
})`
    display: flex;
    margin-left: -12px;
    margin-right: -12px;
    margin-top: -12px;

    .columns:last-child {
        margin-bottom: -12px;
    }
    .columns:not(:last-child) {
        margin-bottom: 12px;
    }
`;

export const Column = styled.div.attrs({
    className: 'Column',
})`
    display: block;
    flex-basis: 0;
    flex-grow: 1;
    flex-shrink: 1;
    padding: 12px;
`;
