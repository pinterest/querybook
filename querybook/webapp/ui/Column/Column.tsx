import styled from 'styled-components';

export const Columns = styled.div.attrs({
    className: 'Columns',
})`
    display: flex;
    margin-left: -12px;
    margin-right: -12px;
    margin-top: -8px;

    .columns:last-child {
        margin-bottom: -8px;
    }
    .columns:not(:last-child) {
        margin-bottom: 8px;
    }
`;

export const Column = styled.div.attrs({
    className: 'Column',
})`
    display: block;
    flex-basis: 0;
    flex-grow: 1;
    flex-shrink: 1;
    padding: 12px;
`;
