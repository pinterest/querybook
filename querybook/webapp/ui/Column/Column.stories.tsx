import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Column, Columns } from './Column';

export default {
    title: 'Layout/Column',
    decorators: [centered],
};
export const _Column = () => (
    <Columns>
        <Column>First Column</Column>
        <Column>Second Column</Column>
        <Column>Third Column</Column>
    </Columns>
);
