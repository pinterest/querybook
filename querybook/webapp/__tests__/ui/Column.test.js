import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { Columns, Column } from '../../ui/Column/Column';

it('renders without crashing', () => {
    shallow(
        <Columns>
            <Column>Test</Column>
            <Column>Test</Column>
        </Columns>
    );
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(
            <Columns>
                <Column>Test</Column>
                <Column>Test</Column>
            </Columns>
        );
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(
            <Columns>
                <Column>Test</Column>
                <Column>Test</Column>
            </Columns>
        );
        expect(output).toMatchSnapshot();
    });
});
