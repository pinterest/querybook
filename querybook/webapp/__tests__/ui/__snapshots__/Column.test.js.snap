// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<div
  className="Column__Columns-sl2jp4-0 gOWWVh Columns"
>
  <Column>
    Test
  </Column>
  <Column>
    Test
  </Column>
</div>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="Column__Columns-sl2jp4-0 gOWWVh Columns"
>
  <div
    className="Column-sl2jp4-1 gozZXJ Column"
  >
    Test
  </div>
  <div
    className="Column-sl2jp4-1 gozZXJ Column"
  >
    Test
  </div>
</div>
`;
